frontend/www/js/omegaup/components/login/Signup.test.ts

```diff
@@ -0,0 +1,242 @@
+  const now = new Date('2022-01-01T00:00:00Z').getTime();
+  let dateNowSpy: jest.SpyInstance<number, []> | null = null;
+  beforeEach(() => {
+    dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => now);
+  });
+  afterEach(() => {
+    if (dateNowSpy) {
+      dateNowSpy.mockRestore();
+    }
+  });
+  const expectedValues: {
+    username: string;
+    email: null | string;
+    parent_email: null | string;
+    password: string;
+    recaptcha: string;
+    birth_date: Date;
+  } = {
+    username: 'Omegaup',
+    email: null,
+    parent_email: 'parentEmail@gmail.com',
+    password: 'pass12345678',
+    recaptcha: '',
+    birth_date: new Date('2012-01-01'),
+  };
+
+  it('should handle register form with initial values', () => {
+    const wrapper = mount(login_Signup, {
+      propsData: {
+        validateRecaptcha: false,
+      },
+    });
+
+    // All the fields in the form are disabled by default
+    expect(
+      wrapper.find('input[name="reg_username"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_email"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_password"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_password_confirmation"]').attributes()
+        .disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[type="checkbox"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('button[name="sign_up"]').attributes().disabled,
+    ).toBeTruthy();
+
+    // Except the birthdate field
+    expect(
+      wrapper.find('input[name="reg_birthdate"]').attributes().disabled,
+    ).toBeFalsy();
+  });
+
+  it('should handle an uncomplete registration for Under13 user', async () => {
+    const wrapper = mount(login_Signup, {
+      propsData: {
+        validateRecaptcha: false,
+      },
+    });
+
+    await wrapper.findComponent(omegaup_DatePicker).setValue('2012-01-01');
+
+    // All the fields in the form are initially disabled
+    expect(
+      wrapper.find('input[name="reg_username"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_parent_email"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_password"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_password_confirmation"]').attributes()
+        .disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[type="checkbox"]').attributes().disabled,
+    ).toBeTruthy();
+    expect(
+      wrapper.find('input[name="reg_birthdate"]').attributes().disabled,
+    ).toBeFalsy(); // the birthdate field should be enabled initially
+
+    // Now, email field doesn't exist because user is U13, so they should provide parent's email
+    expect(wrapper.find('input[name="reg_email"]').exists()).toBe(false);
```

Comment:
Now, all the fields should be enabled
```suggestion
    // Once reg_birthdate has been set, all the fields are enabled
    expect(wrapper.find('input[name="reg_birthdate"]').element).toBeEnabled();
    expect(wrapper.find('input[name="reg_username"]').element).toBeEnabled();

    // Now, email field doesn't exist because user is U13, so they should provide parent's email
    expect(wrapper.find('input[name="reg_email"]').exists()).toBeFalsy();

    expect(
      wrapper.find('input[name="reg_parent_email"]').element,
    ).toBeEnabled();
    expect(wrapper.find('input[name="reg_password"]').element).toBeEnabled();
    expect(
      wrapper.find('input[name="reg_password_confirmation"]').element,
    ).toBeEnabled();
    expect(policyPrivacyCheckbox).toBeEnabled();
```